core/domain/suggestion_services_test.py

```diff
@@ -2253,23 +2253,279 @@ def test_update_translation_contribution_stats(self) -> None:
+
+        change_dict = {
+            'cmd': 'add_translation',
+            'content_id': 'content',
+            'language_code': 'hi',
+            'content_html': '',
+            'state_name': 'Introduction',
+            'translation_html': '<p>Translation for content.</p>'
+        }
+        initial_suggestion = suggestion_services.create_suggestion(
+            feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+            feconf.ENTITY_TYPE_EXPLORATION,
+            '0', 1, self.author_id, change_dict, 'description')
+        latest_suggestion = suggestion_services.create_suggestion(
+            feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+            feconf.ENTITY_TYPE_EXPLORATION,
+            '1', 1, self.author_id, change_dict, 'description')
+
+        suggestion_services.update_translation_contribution_stats_at_submission(
+            initial_suggestion, self.author_id, '0'
+        )
+        suggestion_services.update_translation_contribution_stats_at_submission(
+            latest_suggestion, self.author_id, '1'
+        )
+
+        translation_contribution_stats_model = (
+            suggestion_models.TranslationContributionStatsModel.get(
+                'hi', self.author_id, '0'
+            )
+        )
+        assert translation_contribution_stats_model is not None
+        self.assertEqual(
+            translation_contribution_stats_model.submitted_translations_count,
+            2
+        )
+        self.assertEqual(
+            (
+                translation_contribution_stats_model
+                .submitted_translation_word_count
+            ),
+            6
+        )
+        self.assertEqual(
+            translation_contribution_stats_model.accepted_translations_count,
+            0
+        )
+
+        initial_suggestion.status = suggestion_models.STATUS_ACCEPTED
+        latest_suggestion.status = suggestion_models.STATUS_ACCEPTED
+        suggestion_services.update_translation_review_stats(
+            initial_suggestion, self.reviewer_id, '0', True
+        )
+        suggestion_services.update_translation_review_stats(
+            latest_suggestion, self.reviewer_id, '1', True
+        )
+
+        translation_review_stats_model = (
+            suggestion_models.TranslationReviewStatsModel.get(
+                'hi', self.reviewer_id, '0'
+            )
+        )
+
+        assert translation_review_stats_model is not None
+        self.assertEqual(
+            translation_review_stats_model.accepted_translations_count,
+            2
+        )
+        self.assertEqual(
+            (
+                translation_review_stats_model
+                .reviewed_translation_word_count
+            ),
+            6
+        )
+
+        translation_contribution_stats_model = (
+            suggestion_models.TranslationContributionStatsModel.get(
+                'hi', self.author_id, '0'
+            )
+        )
+        assert translation_contribution_stats_model is not None
```

Comment:
Every assert must have a comment justifying it. /cc @sahiljoster32

Reply:
Done